src/Domain/Bootstrap.php

```diff
@@ -130,17 +130,21 @@ function() {
 	 * Define the global feature flag.
 	 */
 	protected function define_feature_flag() {
-		$allowed_flags = [ 'experimental', 'stable' ];
-		$flag          = getenv( 'WOOCOMMERCE_BLOCKS_PHASE' );
-		if ( ! in_array( $flag, $allowed_flags, true ) ) {
-			if ( file_exists( __DIR__ . '/../../blocks.ini' ) ) {
-				$woo_options = parse_ini_file( __DIR__ . '/../../blocks.ini' );
-				$flag        = is_array( $woo_options ) && 'experimental' === $woo_options['woocommerce_blocks_phase'] ? 'experimental' : 'stable';
-			} else {
-				$flag = 'stable';
-			}
-		}
-		define( 'WOOCOMMERCE_BLOCKS_PHASE', $flag );
+		add_action(
+			'plugins_loaded',
+			function() {
+				global $woocommerce_blocks_phase;
+				$allowed_flags = [ '1', '2', '3' ];
+				if ( file_exists( __DIR__ . '/../../blocks.ini' ) ) {
```

Comment:
This always runs; so we can keep this here and then set phase based on context; package or feature plugin.